feat: [CO-1273] allow editing/deleting shared account contact groups

giuliano176 opened the pull request.

frisonisland committed:
feat: remove contact groups from contact groups SecondaryBarView.tsx

- update test to match behavior

giuliano176 committed:
refactor: create dedicated sidebar

giuliano176 committed:
feat: add contact groups to legacy sidebar

- todo: add test to legacy sidebar to show contact groups
- todo: manage routes
- todo: manage api calls

frisonisland committed:
fix: add keys to component in a list

- added test to check Contact Groups is correctly displayed in Sidebar
- fixed keys and style in components because tests were failing (jest Each child in a list should have a unique "key" prop.)
- fixed also "styled should be defined outside component" warning

frisonisland committed:
fix: template literals

- added test to check Contact Groups is not displayed if 'Contacts' folder missing

frisonisland committed:
feat: update contact groups sidebar item onclick route change

giuliano176 committed:
refactor: restructure legacy app-view

frisonisland committed:
chore: contact-groups detail panel - empty-displayer.tsx remove icon

- align tests

frisonisland committed:
feat: add navbar with items count in contact groups

- extract navbar component for item-list view

giuliano176 committed:
refactor: remove randomIndex from detailPanel

giuliano176 committed:
chore: remove unused function

giuliano176 committed:
chore: fix import

giuliano176 committed:
refactor: convert contact-list from jsx to tsx

giuliano176 committed:
chore: graphical alignment between CgList and contacts list

frisonisland committed:
chore: undo empty-dsplayer changes

- empty-displayer still used by DistributionList with icon, made icon optional

frisonisland committed:
chore: general lint-check and fixes on files I touched

- enabled prettier and fixed errors

frisonisland committed:
fix: detail-panel.test.tsx

- brought back detail panel because noticed test was failing
- probably some code relies on specific location, the test passes if we declare routes with /folder, so to avoid any regression I put it back as before
- added folder-list-panel.tsx with similar logic of detail-panel.test.tsx

giuliano176 committed:
feat: add contact-group detail panel to contacts route

frisonisland committed:
feat: use relative path when deleting contact group

- change history to contact-groups relative to current location/module

frisonisland committed:
refactor: move contact groups and distribution-lists in their own folders under views/

- move components together

frisonisland committed:
refactor: rename view

- now distribution list has its own dedicated view

frisonisland committed:
refactor: move app view

- the idea is to keep related things together as for other UI projects
- app: defines entrypoint + contacts and DL modules entrypoint
- contact-groups: defines all contact-groups related views
- distribution-list: same as above but for dl

frisonisland committed:
feat: change create contact-group redirect

- put back app.tsx under src/ (required for build)

giuliano176 committed:
chore: use replaceHistory from useReplaceHistoryCallback

giuliano176 committed:
test(distribution-list-view): remove irrelevant test

giuliano176 committed:
chore: convert breadcrubs from jsx to tsx

giuliano176 committed:
fix: remove redirect to folder 7

giuliano176 committed:
WIP: fix some cg-view tests

giuliano176 committed:
refactor: restructure contacts view routes

frisonisland committed:
feat: change create contact-group redirect

- make displayer-header.tsx logic agnostic
- remove useActiveItem for contact groups
- remove test about internal logic on active item (the test was not clear, it seemed to test specific internal logic rather than behavior)

frisonisland committed:
refactor: active contact group logic relies on parameter

- rely on useParams

frisonisland committed:
test: update displayer-header.test.tsx

- displayer header is now a generic component used to share common UI/display options

frisonisland committed:
fix: cg-view use trimmed path

- trailing / causes issue when testing

frisonisland committed:
test: update cg-view test

- test did not make sense (why use /:id? as Router path), we added it because 'it worked' but actually the component had a trailing / that was causing issues

frisonisland committed:
refactor: change name of find contact groups api response/request + add initial support for shared account

frisonisland committed:
test: add simple test for shared contact groups

- not sure it is the correct query, let's see

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zextras/carbonio-contacts-ui?pullRequest=257) <br/>All committers have signed the CLA.

frisonisland committed:
feat: add account id in contact groups search

- update tests

giuliano176 committed:
feat: add populateSharedContactGroups method to contact-group store

giuliano176 committed:
feat: add contact groups view to shared folders

frisonisland committed:
refactor: rename contact group to contact-group-component.tsx

frisonisland committed:
refactor: define only one findContactGroup api

frisonisland committed:
refactor: define shared cglist view

frisonisland committed:
fix: cg-view route

- misplaced "?" on optional "id" parameter
- yet to add CgListSHared

giuliano176 committed:
feat: contact view group renders item correctly

frisonisland committed:
test: align contact groups view tests with new path

frisonisland committed:
fix: contact group view test

- add fixme, we have to pay attention when redirecting

frisonisland committed:
feat: add cg-displayer for shared account

- while there is a bit of indirection each component is more direct has different behavior, but the same UI. Perhaps we can save some component by effectively using children instead of passing down props.

- bonus: changes are retro-compatible with current solution for main account

giuliano176 committed:
fix: redirect path of delete contact group action

frisonisland committed:
test: update contact-group.test.tsx to match spy replace history

- minor: use constant in cg-displayer.tsx redirect

frisonisland committed:
refactor: move contact groups related stuff under contact-group

- renamed cg => contact-group

frisonisland committed:
fix: useFindShared contact groups to be reactive

- add test for it

giuliano176 committed:
test(contact-group-list): fix test

frisonisland committed:
refactor: move contact group boards

frisonisland committed:
feat: allow editing shared contact group

frisonisland committed:
refactor(contact-groups): let components declare what they want/need

- avoid relying on useParams, pass parameters (ask for item with given id instead of relying on "magic")
- introduce get contact group hooks
- conditionally render displayer (only if id defined, else empty panel)

frisonisland committed:
refactor(contact-groups): rename use active contact group to "use ... from path"

- it must be clear that said hook must be used only if you want/are able to get the contact group using the path id in url, else the hook feels to magical

frisonisland committed:
refactor(contact-groups-actions): extract redirect logic one level up

- redirect is still in useContactGroupActions, so the behavior is still the same. This allows me to move/refactor things slowly and be in control of the changes without making tests explode
- introduced displayer actions header (yet to be used)

frisonisland committed:
refactor(contact-groups-displayer): add empty displayer component

- get rid of displayer component, we can combine components instead of inheriting them, so a combination of <DisplayerWithActions> or <Empty>

frisonisland committed:
refactor(contact-groups-displayer): move displayer logic build

- redirects now are configured at top level

frisonisland committed:
test: add test to fix behavior/correct redirect

frisonisland committed:
chore: close displayers, there are no children to pass

frisonisland committed:
chore: contact group list extract items

giuliano176 committed:
chore: pass items to ContactGroupList as children

giuliano176 committed:
fix: redirect to correct path when new group is created

giuliano176 committed:
feat: introduce useDeleteSharedContactGroup

giuliano176 committed:
fix: add accountId to populateContactGroupByAccountId

frisonisland committed:
WIP: refactor contact group actions

- define shared contact group actions for specific SharedAccount type and behavior
- TODO: contact-group-list-item.test.tsx broken
- check again edit and send mail actions (check which store they are calling)

frisonisland committed:
Merge remote-tracking branch 'origin/feat/CO-1273' into feat/CO-1273

frisonisland committed:
WIP: refactor contact group list

- fixed tests
- fixed useEvaluateAction type must not be optional
- extract main and shared list item component -> tests on item list are retrocompatible

frisonisland committed:
WIP: refactor contact group action usages

- actions need to be evaluated per contact group

frisonisland committed:
refactor: rearranged logic in delete contact groups actions separating component from business logic

- added back snackbar and flow control for delete modal

- old tests fixed automatically

giuliano176 committed:
fix: change signature of populateSharedContactGroups

giuliano176 committed:
test(delet-contact-group): fix test

giuliano176 committed:
test(contact-group-pagination): fix test

giuliano176 committed:
WIP

giuliano176 committed:
feat: add pagination to sharedAccountGroups view

giuliano176 committed:
Merge branch 'devel' into feat/CO-1273

frisonisland committed:
feat: avoid api call if account data already initialized

- add more tests for useFindSharedContactGroups

frisonisland committed:
chore: remove unused contact groups store methods

frisonisland committed:
test: add test about deleting shared contact group from within list

frisonisland committed:
test: move main account tests inside main account block description

frisonisland committed:
test: add correct setup for shared account (url and components differ)

frisonisland committed:
WIP: add test for shared account contact group detail panel

frisonisland committed:
refactor: remove displayer-controller

- prefer relying on routing instead of if/else inside displayer controller
- rename contact group displayer-header to displayer-component. The component is responsible of showing an empty panel or not

frisonisland committed:
test: add test about shared account contact group displayer

giuliano176 committed:
chore: clean up logTestingPlayground

giuliano176 committed:
test(sidebar-accordion): add tests for shared folders

giuliano176 committed:
WIP

frisonisland committed:
feat: shared contact group add active property to list item

frisonisland committed:
test: move test inside "list" describe block

- split tests about list from displayer

frisonisland committed:
test: move test inside "list" describe block also for main account

- align test structure for contact group view

frisonisland committed:
feat: add active also for main account list item in contact groups

- NOTE: code can still be refactored for common scenarios such as displaying item as "active". For example both shared account and main account item use the same "id" from path

frisonisland committed:
chore: forgot to update dependencies of main account list item

frisonisland committed:
test: check only first/main account folders are shown in collapsed sidebar

- use collapsed-sidebar-item.tsx in collapsed-sidebar-folder-item.tsx

giuliano176 committed:
fix: redirect when clicking collapsed contact groups icon

giuliano176 committed:
chore: update commons

giuliano176 committed:
chore: return type of deleteContactGroup action

giuliano176 committed:
refactor: convert selectionInteractive to tsx

giuliano176 committed:
chore: remove unused code

giuliano176 committed:
refactor: reduce cognitive complexity of delete function

giuliano176 committed:
chore: remove unused fragment

giuliano176 committed:
chore: remove unneccessary fragment

giuliano176 committed:
WIP

frisonisland committed:
test: fix test on find shares button

- accordion uses LocalStorage from shell to check which accordion is expanded. If it is collapsed the button is not visible, so we have no option other than mocking the shell

frisonisland committed:
test:add test to check contact groups in sidebar redirects to correct path

frisonisland committed:
test: add test to check contact groups in sidebar redirects to correct path when shared account

giuliano176 committed:
chore: revert changes to selection-interactive

frisonisland committed:
test: collapsed contact group redirect is ok
